src/codegen.cpp

```diff
@@ -6757,8 +6753,8 @@ static Function *emit_tojlinvoke(jl_code_instance_t *codeinst, Module *M, jl_cod
     auto invoke = jl_atomic_load_relaxed(&codeinst->invoke);
     bool cache_valid = params.cache;
 
-    if (cache_valid && invoke != NULL && invoke != &jl_fptr_wait_for_compiled) {
-        StringRef theFptrName = jl_ExecutionEngine->getFunctionAtAddress((uintptr_t)invoke, codeinst);
+    if (cache_valid && invoke != NULL && invoke != jl_fptr_wait_for_compiled_addr) {
+        StringRef theFptrName = jl_ExecutionEngine->getFunctionAtAddress((uintptr_t)invoke, invoke, codeinst);
```

Comment:
Same here wrt jit dependency

Reply:
Since this is only called from jl_compile_workqueue after determining the `invoke` pointer, it looks like we should change this to taking this value as an argument instead